docs/use-keyring-file-vault-components.md

```diff
@@ -0,0 +1,155 @@
+# Use the keyring_file and keyring_vault components
+
+Percona Server supports a keyring that enables internal server components to store sensitive information securely for later retrieval. The server uses a manifest to load the component and the component has its own configuration file. The components are stored in the `plugin` directory.
+
+!!! note
+
+    Only one keyring component can be enabled at a time. Enabling multiple keyring components is not supported.
+
+Percona Server supports the following keyring components:
+
+=== "keyring_vault component"
+
+    ### Use the keyring vault component
+
+    The `keyring_vault` component extends the server capabilities and provides an interface for the database with a [HashiCorp Vault](https://www.hashicorp.com/products/vault/data-protection) server to store key and secure encryption keys.
+
+    !!! admonition "See also"
+
+        [Hashicorp Documentation: Installing Vault]
+        
+        [Hashicorp Documentation: Production Hardening]
+
+    The server uses a manifest to load the component and the component has its own configuration file.
+
+    An example of a manifest and a configuration file is the following:
+
+    An example of `./bin/mysqld.my`:
+
+    ```json
+    {
+       "components": "file://component_keyring_vault"
+    }
+    ```
+
+    An example of `/lib/plugin/component_keyring_vault.cnf`:
+
+    ```json
+    {
+       "path": "/var/lib/c/keyring_vault", "read_only": false
+    }
+    ```
+
+    Set the `keyring_vault_config` variable to point to the `keyring_vault` configuration file.
+
+    The `keyring_vault_config` variable defines the location of the `keyring_vault` configuration file.
+
+    `keyring_vault_config`
+
+    | Option       | Description            |
+    |--------------|------------------------|
+    | Command-line | --keyring-vault-config |
+    | Scope        | Global                 |
+    | Dynamic      | Yes                    |
+    | Data type    | Text                   |
+    | Default      |                        |
+
+    The `keyring_vault_config` file contains the following information:
+
+    * `timeout` - the duration in seconds for the Vault server connection timeout. The default value is 15. The allowed range is from 0 to 86400. The timeout can be also disabled to wait an infinite amount of time by setting this variable to 0.
+
+    * `vault_url` - the Vault server address.
+
+    * `secret_mount_point` - the mount point name where the keyring_vault stores the keys.
+
+    * `token` - a token generated by the Vault server.
+
+    * `secret_mount_point_version [optional]` - the `KV Secrets Engine version (kv or kv-v2)` used.
+
+    * `vault_ca [optional]` - if the machine does not trust the Vault’s CA certificate, this variable points to the CA certificate used to sign the Vault’s certificates.
+
+    ??? example "Example of a configuration file"
+
+        ```text
+        timeout = 15
```

Comment:
The config content must be a valid JSON

'''
{
    "timeout": 15,
    "vault_url": "https://vault.public.com:8202",
    "secret_mount_point": "secret",
    "secret_mount_point_version": "AUTO",
    "token": "58a20c08-8001-fd5f-5192-7498a48eaf20",
    "vault_ca": "/data/keyring_vault_confs/vault_ca.crt"
}
'''

Reply:
fixed